core/src/main/java/com/orientechnologies/orient/core/storage/ridbag/sbtree/OSBTreeCollectionManagerShared.java

```diff
@@ -197,13 +197,11 @@ public boolean tryDelete(OBonsaiCollectionPointer collectionPointer, long delay)
 
       treeCache.remove(cacheKey);
     }
-    OSBTreeBonsai<OIdentifiable, Integer> treeBonsai = this.loadSBTree(collectionPointer);
+    OSBTreeBonsai<OIdentifiable, Integer> treeBonsai = this.loadTree(collectionPointer);
```

Comment:
I still have doubts about change. Let suppose we removed a tree from the cache. Then let suppose that thread is paused there. After that, we load rid bag again, and after that, we delete ridag in the original thread. So we have to delete rid bag inside the cache. Could you confirm or explain the reason why my doubts are not true?

Reply:
yes this can happen, if though quite unlucky, the only solution here is protecting this delete operation with a lock, anyway this lock was not there before.